src/focus-trap-react.js

```diff
@@ -55,24 +60,35 @@ class FocusTrap extends React.Component {
     }
   }
 
-  /** Returns focus to the element that had focus when the trap was activated. */
-  returnFocus() {
-    const { checkCanReturnFocus, onPostDeactivate } =
-      this.props.focusTrapOptions;
+  deactivateTrap() {
+    const { checkCanReturnFocus } = this.tailoredFocusTrapOptions;
+
+    if (this.focusTrap) {
+      // NOTE: we never let the trap return the focus since we do that ourselves
+      this.focusTrap.deactivate({ returnFocus: false });
+    }
 
-    const sendFocus = () => {
-      if (this.previouslyFocusedElement?.focus) {
+    const finishDeactivation = () => {
+      const canReturnFocus =
+        this.previouslyFocusedElement?.focus && this.returnFocusOnDeactivate;
+
+      if (canReturnFocus) {
+        /** Returns focus to the element that had focus when the trap was activated. */
         this.previouslyFocusedElement.focus();
```

Comment:
I think we need to do something similar to https://github.com/focus-trap/focus-trap/blob/master/index.js#L222 here, because we're ignoring the `setReturnFocus` option (as we were before; this is an existing bug).